### ISSUE TITLE
Powershell invocation is missing arguments

### ISSUE THREAD

**Issue by glennsarti** [Microsoft/vscode-test#2]
The powershell invocation at;

```
			cp.spawnSync('powershell.exe', [
				'-Command',
				`Expand-Archive -Path ${vscodeArchivePath} -DestinationPath ${extractDir}`
```

is missing some arguments which you should use to avoid some edge cases:

`-NoProfile` - Stops loading custom users profile scripts.  This also has a benefit of speeding up the loading process start time

`-NonInteractive` - Suppress any UI questions.  You have no UI at this point

`-NoLogo` - May as well suppress the logo while you're at it

- Probably don't need to set the execution policy (https://docs.microsoft.com/en-us/powershell/scripting/components/console/powershell.exe-command-line-help?view=powershell-5.0)

- Also note that exit codes from powershell.exe, if you decide to use them, could be different from what you expect as ErrorAction has not been set

- Both `${vscodeArchivePath}`  and `${extractDir}` could contain spaces in the path so these should be wrapped in double quotes

- `Expand-Archive` is *ONLY* available in PowerShell 5.0 and above.  Moving from the old gulp method in [vscode-extension-vscode](https://github.com/Microsoft/vscode-extension-vscode/compare/1.1.30...v1.1.32#diff-f900621cc6a1719b5473baf0191f297bR12) to this could be considered a breaking change as Server 2012 and Windows 7 users will not have Expand-Archive by default.  At the *VERY* least this should be a documentation note for Extension Authors.

Ref - https://github.com/Microsoft/vscode-test/blob/master/lib/download.ts#L90-L92

**Comment by glennsarti** [comment]
Ping @rkeithhill @rjmholt @SeeminglyScience @TylerLeonhardt for visibility

**Comment by rkeithhill** [comment]
Also, there are multiple implementations of `Expand-Archive` out there with different parameter names.  I recommend invoking it like so `Microsoft.PowerShell.Archive\Expand-Archive` to ensure you get the built-in version.  Well, for PS >= v5.

**Comment by octref** [comment]
Thanks for the help, I'm not very familiar with Powershell.

> -NoProfile - Stops loading custom users profile scripts. This also has a benefit of speeding up the loading process start time
> -NonInteractive - Suppress any UI questions. You have no UI at this point
> -NoLogo - May as well suppress the logo while you're at it

Added.

> Probably don't need to set the execution policy

I didn't. Should I?

> Also note that exit codes from powershell.exe, if you decide to use them, could be different from what you expect as ErrorAction has not been set

I'm now catching all errors and throwing my own errors.

> Both ${vscodeArchivePath} and ${extractDir} could contain spaces in the path so these should be wrapped in double quotes

Added.

> Expand-Archive is ONLY available in PowerShell 5.0 and above.

Added notice to README. I think the main usage of this is to run it on Azure DevOps hosted agents, and all Windows Agents have PS > 5.0, so this should be fine.

> Microsoft.PowerShell.Archive\Expand-Archive to ensure you get the built-in version.

Done.

I also added a [Changelog](https://github.com/Microsoft/vscode-test/blob/master/CHANGELOG.md) and definitions for Azure DevOps and Travis at https://github.com/octref/vscode-test-sample.

**Comment by octref** [comment]
Feel free to continue replying, just closing since majority of the concerns are addressed.
